### PR TITLE
Added fakery to the various database operations.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Most methods of the guestagent now have fakes.
- The Servers fakes now cache the context to simulate account permissions.

### CODE REVIEW

**Comment by rnirmal** [comment]
+1

**Comment by hub-cap** [comment]
+1

File: reddwarf/tests/fakes/nova.py [hunk 1]
```diff
@@ -91,7 +96,7 @@ def __init__(self, parent, id, name, image_id, flavor_ref):
 
     @property
     def addresses(self):
-        return ["We don't even use this."]
+        return {"private":[{"addr":"123.123.123.123"}]}
```
Comment (ed-) [inline reddwarf/tests/fakes/nova.py:99]: Looks legit

**Comment by ed-** [comment]
+:1:

**Pull request merged**